### ISSUE TITLE
FR: support `accept-dns` in Android client

### ISSUE THREAD

**Issue by Aaroneisele55** [tailscale/tailscale#7311]
### What are you trying to do?

I would like to be able to configure different dns servers per tailscale node in the admin console, because on other platfors there is the --accept-dns option, which is not available on Android or iOS.

### How should we solve this?

Maybe let users choose which node, user or tag  (in DNS settings) should use which DNS server or add a control similar to the `--accept-dns` cli option

### What is the impact of not solving this?

I have one central Raspberry where NodeRed, PiHole and Tailscale run at the same time. That raspberry is also an exit node and subnet router for my network.

I've set the tailscale network DNS to my pihole address (with "override local DNS Settings" enabled), and that forces everything DNS related to go to the PiHole. 
I now have one device that **should** use the subnet router feature, but **shouldn't** use the DNS of PiHole.

My first approach was sharing the Raspberry into a separate network where only that special device is in, but sharing doesn't support the subnet router feature.

**Comment by DentonGentry** [comment]
> I can only use my subnet router in one net, but want some devices to use that RPi (where PiHole is also running) as DNS and some to not to use it.
> All devices should be able to use the routes exposed by it, and this isn't possible with the "share" feature yet.

So is the actual problem in not being able to make the DNS server available on multiple tailnets?

For this specific case there probably is a way to make it work by using userspace-networking to join a second tailnet. The main tailscaled running from systemd would join one tailnet, and then you could write scripts to start others:

```
mkdir /var/lib/tailscale2 
tailscaled --socket=/var/lib/tailscale2/sock --statedir=/var/lib/tailscale2 --tun=userspace-networking
tailscale --socket=/var/lib/tailscale2/sock up
```

userspace-networking's main drawback is that outgoing connections are harder, apps must be configured to use a SOCKS5 or HTTP proxy. For a DNS server which really just receives incoming packets, userspace-networking will receive the incoming packets and send them over to the same port on localhost.

**Comment by Aaroneisele55** [comment]
> So is the actual problem in not being able to make the DNS server available on multiple tailnets?

No, the problem is to tell one node to not to use the DNS i configured in DNS settings in tailscale, and instead use a public on or its default

**Comment by amanjuman** [comment]
This is one of the significant issues in my country. When you use Tailscale with the mobile network provider, they do not let you use any exit node. Sometimes direct exit node works fine but most of the time not. They do not allow to use any other DNS server when Tailscale forces them to use 100.100.100.100. As a result, there is no internet connectivity.

**Comment by agottardo** [comment]
The new version of the app (shipping very soon) includes this.

Implemented in tailscale/tailscale-android#233.
